frontend/packages/noobaa-storage-plugin/src/components/bucket-class/wizard-pages/backingstore-page.tsx

```diff
@@ -0,0 +1,242 @@
+};
+
+const getNames = (list: K8sResourceKind[]) => {
+  return list.reduce((acc, bs) => {
+    const obj = {
+      cells: [],
+      selected: false,
+      id: bs.metadata.name,
+      selectedBy: '',
+    };
+    const type = typeMappings[_.get(bs, ['spec', 'type'])];
+    obj.cells = [
+      {
+        title: (
+          <ResourceLink
+            kind={NooBaaBackingStoreModel.kind}
+            name={bs.metadata.name}
+            title={bs.metadata.uid}
+          />
+        ),
+      },
+      {
+        title: _.get(bs, ['spec', type, 'targetBucket']),
+      },
+      {
+        title: type,
+      },
+      {
+        title: _.get(bs, ['spec', type, 'region']),
+      },
+    ];
+    return [...acc, obj];
+  }, []);
+};
+
+const filterSelectedItems = (items, tableId) =>
+  items.filter((e) => e.selectedBy === tableId).map((e) => e.id);
+
+const BackingStorePage: React.FC<BackingStorePageProps> = React.memo(
+  ({ backingStores, dispatcher, state }) => {
+    const { data: backingStoreData } = backingStores;
+    const { backingStores: storeMain } = state;
+    const { tier2Policy } = state;
+    const showTier2Table = !!tier2Policy;
+    const [searchInput, setSearchInput] = React.useState('');
+    const [searchInput2, setSearchInput2] = React.useState('');
+
+    const filterSearch = (search, list) => {
+      if (!search) return list;
+      return list.filter((elem) => fuzzy(search, elem.id));
+    };
+
+    React.useEffect(() => {
+      if (state.backingStores.length === 0) {
+        const stores = getNames(backingStoreData);
+        dispatcher({ type: 'setBackingStores', value: stores });
+      }
+    }, [backingStoreData, dispatcher, state.backingStores.length]);
+
+    const onSelect = (isSelected, tableId, rowData) => {
+      const selectedItem = storeMain.find((elem) => elem.id === rowData.id);
+      const store = new Set(storeMain);
+      store.delete(selectedItem);
+      selectedItem.selected = !selectedItem.selected;
+      isSelected ? (selectedItem.selectedBy = tableId) : (selectedItem.selectedBy = '');
+      store.add(selectedItem);
+      dispatcher({ type: 'setBackingStores', value: [...store] });
+      const items1 = filterSelectedItems(storeMain, 0);
+      if (items1.length > 0) dispatcher({ type: 'setBackingStoreTier1', value: items1 });
+      const items2 = filterSelectedItems(storeMain, 1);
+      if (items2.length > 0) dispatcher({ type: 'setBackingStoreTier2', value: items2 });
+    };
+
+    const description = React.useCallback(
+      () => (
+        <>
+          <p>
+            BackingStore represents a storage target to be used as the underlying storage for the
+            data in MCG buckets.
+          </p>
+          <p>
+            Multiple types of backing-stores are supported: asws-s3, s3-compatible,
+            google-cloud-storage, azure-blob, obc, PVC.
+          </p>
+        </>
+      ),
+      [],
+    );
```

Comment:
Whats the use case for making it a React Component.
When we can have something like this
```
const description = (
      <>
        <p>
          BackingStore represents a storage target to be used as the underlying storage for the data
          in MCG buckets.
        </p>
        <p>
          Multiple types of backing-stores are supported: asws-s3, s3-compatible,
          google-cloud-storage, azure-blob, obc, PVC.
        </p>
      </>

Reply:
ack